R/outbreak_model.R

```diff
@@ -46,32 +46,24 @@
 #' @importFrom data.table rbindlist
 #'
 #' @examples
-#'
-#'\dontrun{
-#' incfn <- dist_setup(dist_shape = 2.322737,dist_scale = 6.492272)
-#' # delay distribution sampling function
-#' delayfn <- dist_setup(2, 4)
-#' # generate initial cases
-#' case_data <- outbreak_setup(num.initial.cases = 5,
-#'                             incfn=incfn,
-#'                             delayfn = delayfn,
-#'                             k=1.95,
-#'                             prop.asym=0)
-#' # generate next generation of cases
-#' case_data <- outbreak_step(case_data = case_data,
-#'                            disp.iso = 1,
-#'                            disp.com = 0.16,
-#'                            disp.subclin = 0.16,
-#'                            r0isolated = 0,
-#'                            r0community = 2.5,
-#'                            r0subclin = 1.25,
-#'                            prop.asym = 0,
-#'                            incfn = incfn,
-#'                            delayfn = delayfn,
-#'                            prop.ascertain = 0,
-#'                            k = 1.95,
-#'                            quarantine = FALSE)
-#'}
+#' set.seed(1)
+#' out <- outbreak_model(
+#'   num.initial.cases = 1,
+#'   prop.ascertain = 0.2,
+#'   cap_max_days = 350,
+#'   cap_cases = 4500,
+#'   r0isolated = 0.5,
+#'   r0community = 2.5,
+#'   r0subclin = 2.5,
+#'   disp.iso = 1,
+#'   disp.com = 0.16,
+#'   disp.subclin = 0.16,
+#'   k = 0,
+#'   delay_shape = 1.65,
+#'   delay_scale = 4.28,
+#'   prop.asym = 0,
+#'   quarantine = FALSE
+#' )
```

Comment:
how should people be looking at `out`?

also: again, pretty noisy - defaults available?

Reply:
`outbreak_model()` only has `NULL` defaults, which don't seem to have any functional purpose. If `outbreak_model()` is called without specifying any arguments then it produces an unhelpful error message:

```r
Error in rweibull(shape = dist_shape, scale = dist_scale, ...) : 
  invalid arguments
```

> how should people be looking at out?

Tbh I'm not sure. It's just a `<data.table>` so it can just be printed.

Reply:
K, seems like the arguments API issue is for another PR to deal with.

Re showing the user something: my point is more that the current approach (assigning to a variable in this and other case) doesn't just print. e.g.
```r
thing <- function(x, y) { dt <- data.table(x = x, y = y); return(dt[]) }
out <- thing(1:5, 5:1) # doesn't print
thing(1:5, 5:1) # prints
```